src/test/java/org/broadinstitute/barclay/argparser/TaggedArgumentTest.java

```diff
@@ -0,0 +1,301 @@
+            Assert.assertEquals(taggedArg.argValue, tagNameValuePair.getValue());
+        }
+
+        // validate attributes
+        i = 0;
+        for (KeyValuePair attributePairArray[] : attributePairArrays) {
+            Map<String, String> attributes = taggable.taggableArgList.get(i++).getTagAttributes();
+            for (KeyValuePair attributePair : attributePairArray) {
+                Assert.assertEquals(attributes.get(attributePair.getKey()), attributePair.getValue());
+            }
+        }
+    }
+
+    @DataProvider(name = "BadTaggedArguments")
+    public Object[][] badTaggedArguments() {
+        return new Object[][]{
+                // long name
+                {new String[]{"--t"}},                              // taggable, but no tagname, no arg
+                {new String[]{"--t:"}},                             // taggable, tagname missing, no arg
+                {new String[]{"--t:", "tumor1.bam"}},               // taggable, tagname missing, with arg
+                {new String[]{"--t:tagName"}},                      // taggable, with tagname, but no arg
+
+                // short name
+                {new String[]{"--tFullName"}},                      // taggable, but no tagname, no arg
+                {new String[]{"--tFullName:"}},                     // taggable, tagname missing, no arg
+                {new String[]{"--tFullName:", "tumor1.bam"}},       // taggable, tagname missing, with arg
+                {new String[]{"--tFullName:tagName"}},              // taggable, with tagname, but no arg
+
+                // not taggable
+                {new String[]{"--n:tagName"}},                      // not taggable, with tagname
+                {new String[]{"--n:key=value"}},                    // not taggable, with attributes
+                {new String[]{"--n:tagName,key=value"}},            // not taggable, with tagname and attributes
+                {new String[]{"--n:tagName,key=value argValue"}},   // not taggable, with tagname, attributes and value
+
+                // missing value
+                {new String[]{"--t:tagName,key=value"}},             // taggable, with tagname, attributes, missing value at end
+                {new String[]{"--t:tagName,key=value", "-s", "28"}}, // taggable, with tagname, attributes, missing value and second argument
+                {new String[]{ "-s", "28", "--t:tagName,key=value"}},// second argument taggable with tagname, attributes, missing value
+
+                // Malformed attribute strings
+                {new String[]{"--t:tumor,truth", "tumor.bam"}},                    // attribute name with missing value
+                {new String[]{"--t:tumor,truth=", "tumor.bam"}},                   // attribute name with missing value
+                {new String[]{"--t:tumor,truth=true,truth=false", "tumor.bam"}},   // duplicate attribute value
+                {new String[]{"--t:tumor,", "tumor.bam"}},                         // dangling comma
```

Comment:
Need a lot of additional test cases in this `DataProvider`:

`--arg:tag:tag value`
`--arg:tag,key=value=foo value`
`--arg:tag,, value`
`--arg:tag,,key=value value`
`--arg:,key=value value`
`--arg:,,key=value value`

etc. etc. See `InvalidFeatureArgumentValuesDataProvider` in `FeatureInputUnitTest` in GATK4, and try to replicate all the test cases there.